CRM/Contact/BAO/Query.php

```diff
@@ -856,7 +840,20 @@ public function selectClause($apiEntity = NULL) {
                 }
               }
               else {
-                $this->_select[$name] = "{$field['where']} as `$name`";
+                // If we have an option group defined then rather than joining the option value table in
+                // (which is an unindexed join) we render the option value on output.
+                // @todo - extend this to other pseudoconstants.
+                if ($this->pseudoConstantNameIsInReturnProperties($field)) {
```

Comment:
By adding all option-value based fields (that get here) to the pseudoconstant array they become workable in the context of the $convertToPseudoFields fn